### PR TITLE
NAS-130884: fix data-source

### PR DESCRIPTION
**Pull request by stavros-k**
**Changes:**

➡️ Feel free to take over and do any changes as you see fit!

Fixing the data-source of the 2 fields, the payload now should include the actual data

Original PR https://github.com/truenas/webui/pull/10673

Example Payload

```json
{
  "id": "9342af35-5bf4-8aba-fd45-793026e5359a",
  "msg": "result",
  "result": [
    {
      "catalog": "TRUENAS",
      "installed": false,
      "train": "stable",
      "app_readme": "<h1>Pi-hole</h1> <p><a href=\"https://pi-hole.net/\">Pi-hole</a> is a black hole for Internet advertisements</p>",
      "categories": ["networking"],
      "description": "DNS and Ad-filtering for your network.",
      "healthy": true,
      "healthy_error": null,
      "home": "https://pi-hole.net",
      "location": "/mnt/.ix-apps/truenas_catalog/trains/stable/pihole",
      "latest_version": "1.1.6",
      "latest_app_version": "2024.07.0",
      "latest_human_version": "2024.07.0_1.1.6",
      "last_update": {
        "$date": 1726939188000
      },
      "name": "pihole",
      "recommended": false,
      "title": "Pihole",
      "maintainers": [
        {
          "email": "dev@ixsystems.com",
          "name": "truenas",
          "url": "https://www.truenas.com/"
        }
      ],
      "tags": ["networking", "dns"],
      "screenshots": [
        "https://media.sys.truenas.net/apps/pihole/screenshots/screenshot1.png"
      ],
      "sources": [
        "https://pi-hole.net/",
        "https://github.com/truenas/charts/tree/master/charts/pihole"
      ],
      "icon_url": "https://media.sys.truenas.net/apps/pihole/icons/icon.png",
      "capabilities": [
        {
          "description": "Pi-hole is able to perform various network-related operations.",
          "name": "NET_ADMIN"
        },
        {
          "description": "Pi-hole is able to access the network.",
          "name": "NET_RAW"
        },
        {
          "description": "Pi-hole is able to bind to a privileged port.",
          "name": "NET_BIND_SERVICE"
        },
        {
          "description": "Pi-hole is able to chown files.",
          "name": "CHOWN"
        },
        {
          "description": "Pi-hole is able to bypass permission checks for it's sub-processes.",
          "name": "FOWNER"
        },
        {
          "description": "Pi-hole is able to bypass permission checks.",
          "name": "DAC_OVERRIDE"
        },
        {
          "description": "Pi-hole is able to bypass permission checks.",
          "name": "SETGID"
        },
        {
          "description": "Pi-hole is able to bypass permission checks.",
          "name": "SETUID"
        },
        {
          "description": "Pi-hole is able to set the capabilities.",
          "name": "SETPCAP"
        },
        {
          "description": "Pi-hole is able to set the capabilities on a file.",
          "name": "SETFCAP"
        },
        {
          "description": "Pi-hole is able to kill processes.",
          "name": "KILL"
        },
        {
          "description": "Pi-hole is able to set the nice level on a process.",
          "name": "SYS_NICE"
        }
      ],
      "run_as_context": [
        {
          "description": "Pi-hole runs as root user.",
          "gid": 0,
          "group_name": "root",
          "uid": 0,
          "user_name": "root"
        }
      ]
    },
  ],
}
```

**Testing:**

### Downstream

|Affects         |Reasoning
|----------------|-------------------------------
|Documentation   |

### CODE REVIEW

File: src/app/interfaces/available-app.interface.ts [hunk 1]
```diff
@@ -10,6 +10,8 @@ export interface AvailableApp {
   description: string;
   app_readme: string;
   app_metadata: AppMetadata;
```
Comment (stavros-k) [inline src/app/interfaces/available-app.interface.ts:12]: Not sure if the `app_metadata` exists on the "available app".
If this type is used in the `app.available` endpoint, it should probably needs to be removed.
I've attached a sample of one app

Comment (AlexKarpov98) [reply src/app/interfaces/available-app.interface.ts]: then, let's remove `app_metadata: AppMetadata;` from `AvailableApp`.
It'll not break the app, it's not used anywhere.

**Review by AlexKarpov98 — APPROVED** [review]
Looks good, I've updated PR a bit as well.

**Comment by AlexKarpov98** [comment]
backport

**Comment by AlexKarpov98** [comment]
backport

**Pull request merged**